### PR TITLE
SQLite3 mem usage metric

### PR DESCRIPTION
**Pull request by stefantalpalaru**

### CODE REVIEW

File: eth/db/kvstore_sqlite3.nim [hunk 1]
```diff
@@ -3,8 +3,8 @@
 {.push raises: [Defect].}
 
 import
-  os,
-  sqlite3_abi,
+  os, tables, times,
```
Comment (arnetheduck) [inline eth/db/kvstore_sqlite3.nim:6]: these can go under `when defined metrics`?

File: eth/db/kvstore_sqlite3.nim [hunk 2]
```diff
@@ -162,3 +162,35 @@ proc init*(
...
+  var sqlite3Info* {.global.} = newSqlite3Info("sqlite3_info", "SQLite3 info")
+
+  method collect*(collector: Sqlite3Info): Metrics =
+    result = initOrderedTable[Labels, seq[Metric]]()
+    result[@[]] = @[]
+    var
+      timestamp = getTime().toMilliseconds()
+      currentMem, highwaterMem: cint
+
+    if (let res = sqlite3_status(SQLITE_STATUS_MEMORY_USED, currentMem.addr, highwaterMem.addr, 0); res != SQLITE_OK):
```
Comment (arnetheduck) [inline eth/db/kvstore_sqlite3.nim:186]: `sqlite3_status64`

File: eth/db/kvstore_sqlite3.nim [hunk 3]
```diff
@@ -162,3 +162,35 @@ proc init*(
...
+                        creationThreadId: getThreadId())
+    result.register(registry)
+
+  var sqlite3Info* {.global.} = newSqlite3Info("sqlite3_info", "SQLite3 info")
+
+  method collect*(collector: Sqlite3Info): Metrics =
+    result = initOrderedTable[Labels, seq[Metric]]()
+    result[@[]] = @[]
+    var
+      timestamp = getTime().toMilliseconds()
```
Comment (arnetheduck) [inline eth/db/kvstore_sqlite3.nim:183]: not a `var`

File: eth/db/kvstore_sqlite3.nim [hunk 4]
```diff
@@ -162,3 +162,35 @@ proc init*(
...
+    result = Sqlite3Info(name: name,
+                        help: help,
+                        typ: "gauge",
+                        creationThreadId: getThreadId())
+    result.register(registry)
+
+  var sqlite3Info* {.global.} = newSqlite3Info("sqlite3_info", "SQLite3 info")
+
+  method collect*(collector: Sqlite3Info): Metrics =
+    result = initOrderedTable[Labels, seq[Metric]]()
```
Comment (arnetheduck) [inline eth/db/kvstore_sqlite3.nim:180]: this is pretty strange as far as interfaces go - why is it an ordered table over lists of labels??

Comment (stefantalpalaru) [reply eth/db/kvstore_sqlite3.nim:183]: https://prometheus.io/docs/instrumenting/writing_clientlibs/#labels

The order comes in handy when debugging.

File: eth/db/kvstore_sqlite3.nim (hunk 1 above)
Comment (stefantalpalaru) [reply eth/db/kvstore_sqlite3.nim]: Moved.

File: eth/db/kvstore_sqlite3.nim (hunk 2 above)
Comment (stefantalpalaru) [reply eth/db/kvstore_sqlite3.nim]: Replaced.

File: eth/db/kvstore_sqlite3.nim (hunk 3 above)
Comment (stefantalpalaru) [reply eth/db/kvstore_sqlite3.nim]: Changed into a `let`.

**Comment by zah** [comment]
Merged in https://github.com/status-im/nim-eth/pull/237

**Pull request closed without merging**